src/ol_infrastructure/applications/jupyterhub/deployment.py

```diff
@@ -0,0 +1,431 @@
+            resource_suffix="ol-shared-plugins",
+            k8s_namespace=namespace,
+            k8s_labels=application_labels,
+            enable_defaults=True,
+        ),
+    )
+
+    # Certificate
+    api_tls_secret_name = f"api-{base_name}-tls-pair"
+    OLCertManagerCert(
+        f"ol-{base_name}-cert-manager-certificate-{stack_info.env_suffix}",
+        cert_config=OLCertManagerCertConfig(
+            application_name=base_name,
+            k8s_namespace=namespace,
+            k8s_labels=application_labels,
+            create_apisixtls_resource=True,
+            apisixtls_ingress_class=apisix_ingress_class,
+            dest_secret_name=api_tls_secret_name,
+            dns_names=[domain_name],
+        ),
+    )
+
+    # OIDC Configuration
+    # ol-k8s-apisix-olapisixoidcresources-ci
+    # It wants to tear this down since the name is changing. Is that okay?
+    oidc = OLApisixOIDCResources(
+        f"ol-k8s-apisix-{base_name}-olapisixoidcresources-{stack_info.env_suffix}",
+        oidc_config=OLApisixOIDCConfig(
+            application_name=base_name,
+            k8s_labels=application_labels,
+            k8s_namespace=namespace,
+            oidc_logout_path="hub/logout",
+            oidc_post_logout_redirect_uri=f"https://{domain_name}/hub/login",
+            oidc_session_cookie_lifetime=60 * 20160,
+            oidc_use_session_secret=True,
+            oidc_scope="openid email",
+            vault_mount="secret-operations",
+            vault_path="sso/mitlearn",
+            vaultauth=vault_k8s_resources.auth_name,
+        ),
+    )
+
+    # APISIX Route
+    OLApisixRoute(
+        name=f"ol-{base_name}-k8s-apisix-route-{stack_info.env_suffix}",
+        k8s_namespace=namespace,
+        k8s_labels=application_labels,
+        ingress_class_name=apisix_ingress_class,
+        route_configs=[
+            OLApisixRouteConfig(
+                route_name=base_name,
+                priority=0,
+                shared_plugin_config_name=shared_plugins.resource_name,
+                plugins=[
+                    oidc.get_full_oidc_plugin_config(unauth_action="auth"),
+                ],
+                hosts=[domain_name],
+                paths=["/*"],
+                backend_service_name="proxy-public",
+                backend_service_port="http",
+                websocket=True,
+            ),
+        ],
+        opts=ResourceOptions(
+            delete_before_replace=True,
+        ),
+    )
+
+    # JupyterHub Helm Release
+    extra_images_list = extra_images or {}
+    admin_users_list = jupyterhub_config.get_object("admin_users", default=[])
+    allowed_users_list = jupyterhub_config.get_object("allowed_users", default=[])
+    return kubernetes.helm.v3.Release(
+        f"{base_name}-{env_name.upper()}-application-helm-release",
+        kubernetes.helm.v3.ReleaseArgs(
+            chart="jupyterhub",
+            cleanup_on_fail=True,
+            version="4.2.0",
```

Comment:
We should pull this value from `src/bridge/lib/versions.py` so that it's managed by renovate